commandapi-core/src/main/java/dev/jorel/commandapi/CommandAPI.java

```diff
@@ -260,7 +261,18 @@ public static void reloadDatapacks() {
 	 *         {@link CommandSyntaxException}
 	 */
 	public static WrapperCommandSyntaxException fail(String message) {
-		return new WrapperCommandSyntaxException(new SimpleCommandExceptionType(new LiteralMessage(message)).create());
+		return fail(new LiteralMessage(message));
+	}
+
+	/**
+	 * Forces a command to return a success value of 0
+	 *
+	 * @param message Description of the error message, formatted as a brigadier message
+	 * @return a {@link WrapperCommandSyntaxException} that wraps Brigadier's
+	 *         {@link CommandSyntaxException}
+	 */
+	public static WrapperCommandSyntaxException fail(Message message) {
```

Comment:
Adding `CommandAPI.fail` overloads for `BaseComponent[]` and Adventure's `Component` objects is preferred for completeness and consistency (and it's also currently being requested by #334!)

Reply:
Added in d10438a

Reply:
I've observed some compilation errors with the compiler not being able to determine the intended method overload. This is what lead to `Tooltip#ofString` which removes the need for the compiler to determine the appropriate method.

Would it be better if`CommandAPI#fail` were deprecated and replaced with `CommandAPI#failWithString(String)`, `CommandAPI#failWithMessage(Message)`, `CommandAPI#failWithBaseComponents(BaseComponent[])`, and `CommandAPI#failWithAdventureComponent` to avoid this issue too?

Reply:
> I've observed some compilation errors with the compiler not being able to determine the intended method overload. This is what lead to `Tooltip#ofString` which removes the need for the compiler to determine the appropriate method.
> 
> Would it be better if`CommandAPI#fail` were deprecated and replaced with `CommandAPI#failWithString(String)`, `CommandAPI#failWithMessage(Message)`, `CommandAPI#failWithBaseComponents(BaseComponent[])`, and `CommandAPI#failWithAdventureComponent` to avoid this issue too?

That's absolutely fine :)

Reply:
227f1d1